aip/auth/4118.md

```diff
@@ -0,0 +1,144 @@
+---
+id: 4118
+scope: auth
+state: draft
+created: 2022-09-09
+---
+
+# Mutual Authentication Using Workload Credentials
+
+Mutual TLS (a.k.a mTLS) authentication enables authentication of both client and
+server identities in a TLS handshake. With workload credentials, applications
+running in Google Cloud can authenticate to Google APIs using [X.509 SPIFFE
+Verifiable Identity Documents][0] (SVIDs). These SVIDs are X.509 certificates
+that contain [SPIFFE IDs][1] specifying the identity of the certificate owners.
+mTLS authentication using X.509 SVIDs occurs when either the client or the
+server (the client in the case of this AIP) uses an X.509 SVID when performing
+the TLS handshake.
+
+**Note:** Because this AIP describes guidance and requirements in a
+language-neutral way, it uses generic terminology which may be imprecise or
+inappropriate in certain languages or environments.
+
+## Guidance
+
+If users enable token binding, they **should** do so via [ADC][2]. This section
+describes the general guidance of supporting such authentication.
+
+### Provisioning Workload Credentials in Google Cloud
+
+On Google Cloud, workload credentials should be provisioned using one of the
+following methods:
+
+  - [Set up service security with Envoy][3].
+  - [Set up service security with proxyless gRPC][4].
+
+In order for workload credentials to be properly used, the auth libraries
+**must** support the automatic switching of the service endpoint to its mTLS
+counterpart.
+
+### Using Workload Credentials
+
+Users **should** configure [ADC][2] to use workload credentials via the
+certificate configuration gcloud metadata file. Workload credentials can be
+added as a **"cert_info"** type as follows:
+
+```json
+{
+  “version”: 1
+  "cert_info": {
```

Comment:
"cert_info" has been renamed to "cert_configs" in the latest proposal (just shared you a doc on this.)

Reply:
Thanks for letting me know. Fixed.